portfolio/src/main/webapp/user-info/user-info.component.js

```diff
@@ -0,0 +1,29 @@
+angular.module('userInfo').component('userInfo', {
+
+  templateUrl: 'user-info/user-info.template.html',
+  controller: function UserInfoController($http) {
+    var vm = this;
+    vm.loading = true;
+    vm.userInfo = '';
+
+    $http.get('user-info').then(function(response) {
+      vm.userInfo = response.data;
+      vm.loading = false;
+    });
+
+    vm.isLoggedIn = function() {
+      if (!vm.userInfo.includes('_ah/login')) {
+        if (!vm.userInfo == '') {  // TODO: input validation on nicknames so
+                                   // can't have empty string
+          vm.nicknameSet = true;
```

Comment:
All of this logic could also be simpler if we were to return the entire user object. 

For example, returning the whole object would mean we could access fields in our frontend code as 'vm.userInfo.nickname' or 'vm.userInfo.email', etc. 
Then we could get rid of any of the nicknameSet logic here and in our HTML template just have: 'ng-show="userInfoCtrl.userInfo.nickname"'. 

Take a look at this too for why I think we can just use userInfoCtrl.userInfo.nickname in ng-show: https://stackoverflow.com/questions/5515310/is-there-a-standard-function-to-check-for-null-undefined-or-blank-variables-in